### ISSUE TITLE
fix: Listbox and Select have non-native initial selection behaviors

### ISSUE THREAD

**Issue by atmgrifter00** [microsoft/fast#5724]
# 🐛 Bug Report

Both the `fast-listbox` and the `fast-select` have non-native behaviors for displaying an initially selected value.

### Listbox

If the `fast-listbox` is meant to behave like the native `select` element (when its `size` has been set to greater than 1), then if no contained option has been given the `selected` attribute, then the `select` element should show nothing as selected by default, as follows:

![image](https://user-images.githubusercontent.com/26874831/157547505-d69817b8-3722-4e40-b8fc-35562bac0804.png)

Currently the `fast-listbox` will select the first option if no option has been explicitly marked as `selected`:

![image](https://user-images.githubusercontent.com/26874831/157548118-339d85e8-8e55-458c-83d3-b79dcc5b3188.png)

### Select

This [PR](https://github.com/microsoft/fast/issues/4137) made it so that a disabled `fast-select` would not set its value, even to an option that is marked as `selected`. Note that the native behavior does not care about the `disabled` state of the `select`.

#### Native:

![image](https://user-images.githubusercontent.com/26874831/157549171-bc085da9-01bb-4e48-a2d9-077bf5ce15fe.png)

![image](https://user-images.githubusercontent.com/26874831/157549310-8595023d-1495-417b-896b-333e1a2ed77d.png)

#### FAST:

![image](https://user-images.githubusercontent.com/26874831/157549834-635440f5-5d60-4ba0-ac97-9c13fdeaf811.png)

![image](https://user-images.githubusercontent.com/26874831/157549977-958ae6d4-44a2-4ba9-81e8-3d9affe18708.png)

## 💻 Repro or Code Sample

See above.

## 🤔 Expected Behavior

The `fast-listbox` and `fast-select` component should behave like their native counterparts for default selected items.

## 😯 Current Behavior

Outlined above.

## 💁 Possible Solution

The fix seems rather trivial. Revert the change to `setSelectedOptions()` in `listbox.ts` that added the `!this.disabled` check (solves the `fast-select` issue), and remove the assignment of `this.selectedIndex = 0;` as the last statement in `setDefaultSelectedOption()` (solves the `fast-listbox` issue).

I intend to create a PR to this effect.

## 🔦 Context

I came across this issue because we tried to uptake the most recent version of `fast-foundation` and our Chromatic regression checks caught that our disabled `select` components no longer had an initial value. As somewhat of an aside, I'm curious if the FAST team has considered tooling like Chromatic for similar purposes that might have discovered what seems to be an unexpected regression in the `fast-select`.

The broader issue, of course, is that the FAST behaviors aren't aligned with native behaviors, and I'm unsure if there is clear reasoning to stray from them. I would assume the default is to mimic the native behaviors when possible.

## 🌍 Your Environment

* OS & Device: Windows 10
* Browser: Google Chrome

**Comment by chrisdholt** [comment]
Thanks for the issue @atmgrifter00 - lots to address here, so let's start on the possibility of a regression around the PR you noted. Tagging @eljefe223 who created that and @radium-v who has done a lot of work on Select, Listbox, Combobox to take a look at what you've called out as far as regressions in behavior for Select.

As far as native elements, regarding listbox intending to be a select - it's not, they are in fact different things. In native selects, the composition is abstracted and I believe that the Shadow DOM of those custom elements is currently not exposed. A select includes a listbox but is not itself a listbox. For our elements, because we don't yet have customizable controls (which are being incubated in Open UI in partnership with Browser vendors) we need to create these in coordination with the ARIA practices documentation which enumerate expectations around these controls (the spec, not the examples). We've followed guidance enumerated for [listbox](https://w3c.github.io/aria-practices/#Listbox), as well as [Combobox](https://w3c.github.io/aria-practices/#combobox) which single-select is enumerated under as well.

As far as a possible solution for the behavior/regression, happy to take a PR for this, we'll just want to ensure that it works in all three scenarios - listbox, select, combobox. So for listbox, we do need a scenario where a listbox can be set as disabled outside of select, because that's a very real scenario.

We'd welcome a contribution here!

**Comment by atmgrifter00** [comment]
Just so we're all on the same page, I didn't mean to suggest that the native listbox was actually a select or vice versa, only on the expected behaviors of either as demonstrated in the examples. I also recognize that FAST (and our own team) are trying to adhere to ARIA best practices. From what I can read on the listbox (from what you linked) there is no mention that the listbox should default to showing the first item in the list selected, except in the following (under 'Keyboard Interaction'):

"When a single-select listbox receives focus:
- If none of the options are selected before the listbox receives focus, the first option receives focus. Optionally, the first option may be automatically selected."

So only in the case where a listbox receives keyboard focus should the control _optionally_ select the first option. Even the example listbox linked above that section ([Scrollable Listbox Example](https://w3c.github.io/aria-practices/examples/listbox/listbox-scrollable.html)) has no item selected by default.

Also, you mention that you "need a scenario where a listbox can be set as disabled outside of select". I do not disagree with that, and I don't believe the issues I raised relate to the `disabled` behavior of `fast-listbox`. In fact, with the suggested changes I made, I don't see any issue with the disabled `fast-listbox`. It simply doesn't show a default selected item.

Sorry if I'm still not understanding the reasoning behind the current behaviors, or if I missed the points you were raising.

**Comment by chrisdholt** [comment]
> Just so we're all on the same page, I didn't mean to suggest that the native listbox was actually a select or vice versa, only on the expected behaviors of either as demonstrated in the examples. I also recognize that FAST (and our own team) are trying to adhere to ARIA best practices. From what I can read on the listbox (from what you linked) there is no mention that the listbox should default to showing the first item in the list selected, except in the following (under 'Keyboard Interaction'):
> 
> "When a single-select listbox receives focus:
> 
> * If none of the options are selected before the listbox receives focus, the first option receives focus. Optionally, the first option may be automatically selected."
> 
> So only in the case where a listbox receives keyboard focus should the control _optionally_ select the first option. Even the example listbox linked above that section ([Scrollable Listbox Example](https://w3c.github.io/aria-practices/examples/listbox/listbox-scrollable.html)) has no item selected by default.
> 
> Also, you mention that you "need a scenario where a listbox can be set as disabled outside of select". I do not disagree with that, and I don't believe the issues I raised relate to the `disabled` behavior of `fast-listbox`. In fact, with the suggested changes I made, I don't see any issue with the disabled `fast-listbox`. It simply doesn't show a default selected item.
> 
> Sorry if I'm still not understanding the reasoning behind the current behaviors, or if I missed the points you were raising.

Not providing reasoning, just background overall. I think @radium-v mentioned this looks like a big, and as noted above, we are happy to take a fix here!

**Comment by atmgrifter00** [comment]
@chrisdholt, out of curiosity, would a change that altered the default selected item behavior of listboxes be a major change (semantically)? That is, if the listbox no longer has a default selected item, would that be considered a major version bump for `fast-foundation`?

**Comment by chrisdholt** [comment]
> @chrisdholt, out of curiosity, would a change that altered the default selected item behavior of listboxes be a major change (semantically)? That is, if the listbox no longer has a default selected item, would that be considered a major version bump for `fast-foundation`?

Not a good answer, but it depends on how we approach the problem - my initial hunch is that it _may_, but we are planning some breaking changes now. If the change primarily affects listbox, it may *not* be as egregious and we could possibly do it as a fix. So, maybe - but maybe not. 😆 

I think if we're looking at a potential break there, the biggest thing that I'd suggest is just making the case for the proposed fix with some context on the considerations. That really helps provide context to reviewers, but also to the community as we look at potentially making a change in a minor version. Thanks!
